marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/model/ModelControllerTest.java

```diff
@@ -60,26 +69,114 @@ private void updateModelInfo() {
+            "          \"datatype\" : \"string\",\n" +
+            "          \"collation\" : \"http://marklogic.com/collation/codepoint\"\n" +
+            "        }" +
             "      }\n" +
             "    }}";
 
         try {
             controller.updateModelEntityTypes(objectMapper.readTree(entityTypes), MODEL_NAME);
-        } catch (JsonProcessingException e) {
+        }
+        catch (JsonProcessingException e) {
             throw new RuntimeException(e);
         }
 
+        assertSearchOptionsDeployment();
+        assertPIIFilesDeployment();
+        assertIndexDeployment();
+
         assertEquals("string", loadModel(getHubClient().getFinalClient()).get("definitions").get(MODEL_NAME).get("properties").get("someProperty").get("datatype").asText());
     }
 
     private JsonNode loadModel(DatabaseClient client) {
         return client.newJSONDocumentManager().read("/entities/" + MODEL_NAME + ".entity.json", new JacksonHandle()).get();
     }
+
+    private void assertSearchOptionsDeployment() {
+        DatabaseClient stagingDatabaseClient = getHubClient().getStagingClient();
+        DatabaseClient finalDatabaseClient = getHubClient().getFinalClient();
+        Map<String, DatabaseClient> clientMap = new HashMap<>();
+        clientMap.put("staging", stagingDatabaseClient);
+        clientMap.put("final", finalDatabaseClient);
+        clientMap.forEach((databaseKind, databaseClient) -> {
+            QueryOptionsManager queryOptionsManager = databaseClient.newServerConfigManager().newQueryOptionsManager();
+            assertTrue(queryOptionsManager.readOptionsAs("exp-" + databaseKind + "-entity-options", Format.XML, String.class).contains(MODEL_NAME));
+            assertTrue(queryOptionsManager.readOptionsAs(databaseKind + "-entity-options", Format.XML, String.class).contains(MODEL_NAME));
+        });
+    }
+
+    private void assertPIIFilesDeployment() {
+        runAsAdmin();
+        ManageClient manageClient = getHubClient().getManageClient();
+
+        try {
+            assertTrue(manageClient.getJson("/manage/v2/protected-paths").contains("someProperty"));
+
+            JsonNode queryRolesets = objectMapper.readTree(manageClient.getJson("/manage/v2/query-rolesets"));
+            assertTrue(queryRolesets.get("query-roleset-default-list").get("list-items").get("list-count").get("value").asInt() >= 1);
+        }
+        catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void assertIndexDeployment() {
+        ManageClient manageClient = getHubClient().getManageClient();
+        Stream.of(getHubConfig().getDbName(DatabaseKind.STAGING), getHubConfig().getDbName(DatabaseKind.FINAL)).forEach(databaseKind -> {
+            String indexConfig = manageClient.getJson("/manage/v2/databases/" + databaseKind + "/properties");
+            assertTrue(indexConfig.contains("testRangeIndexForDHFPROD4704"));
```

Comment:
A testing technique - a problem with assertTrue/assertFalse is that if it fails, it's not known why. We really want to see the value of indexConfig to find out what's going on. 

To improve on that, I'd do this:

```
["someProperty", "someOtherProperty", etc].forEach(prop -> {
    assertTrue(indexConfig.contains(property), "Expected " + property + " to be in indexConfig: " + indexConfig);
});
```

That makes debugging a test failure much easier.

Reply:
Makes sense. Updated in here b6777af